lib/features/qr/models/seed_response.dart

```diff
@@ -0,0 +1,47 @@
+import 'dart:convert';
+
+import 'package:equatable/equatable.dart';
+
+class SeedResponse extends Equatable {
+  final String? seed;
+  final String? expiresAt;
+  final String? errorMessage;
+
+  const SeedResponse({
+    this.seed,
+    this.expiresAt,
+    this.errorMessage,
+  });
+
+  Map<String, dynamic> toMap() {
+    return {
+      'seed': seed,
+      'expiresAt': expiresAt,
+    };
+  }
+
+  factory SeedResponse.fromMap(Map<String, dynamic> map) {
+    return SeedResponse(
+      seed: map['seed'] != null ? map['seed'] : null,
+       expiresAt: map['expiresAt'] != null ? map['expiresAt'] : null,
+    );
+  }
```

Comment:
I would not need errorMessage for Success response.
I wrote separate factory for exception to set the errorMessage

Reply:
I understand your reasoning, but since you are doing JSON Serialization manually, this method should allow a given `Map<String, dynamic> map` to be turned into a complete `SeedResponse`.